Mathlib/RingTheory/PowerSeries/Substitution.lean

```diff
@@ -0,0 +1,381 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos Fernández
+-/
+
+import Mathlib.LinearAlgebra.Finsupp.Pi
+import Mathlib.RingTheory.MvPowerSeries.Substitution
+import Mathlib.RingTheory.PowerSeries.Evaluation
+import Mathlib.RingTheory.PowerSeries.Basic
+
+import Mathlib.RingTheory.PowerSeries.Order
+
+/-! # Substitutions in power series
+
+A (possibly multivariate) power series can be substituted into
+a (univariate) power series if and only if its constant coefficient is nilpotent.
+
+This is a particularization of the substitution of multivariate power series
+to the case of univariate power series.
+
+-/
+
+namespace PowerSeries
+
+variable
+  {A : Type*} [CommRing A]
+  {R : Type*} [CommRing R] [Algebra A R]
+  {τ : Type*}
+  {S : Type*} [CommRing S]
+
+open MvPowerSeries.WithPiTopology
+
+attribute [local instance] DiscreteTopology.instContinuousSMul
```

Comment:
This is used only once in the file. Can you move it to the relevant declaration (adding `in` at the end)? Thanks!